configuration/src/main/java/io/airlift/configuration/ConfigurationFactory.java

```diff
@@ -130,18 +132,38 @@ public Map<String, String> getProperties()
 
     /**
      * Marks the specified property as consumed.
+     *
+     * @deprecated Use {@link #consumeProperty(ConfigPropertyMetadata)} instead
      */
+    @Deprecated
     public void consumeProperty(String property)
```

Comment:
There is only a single usage of this API in Trino and it's a case when one Bootstrap.configure() is passed to another one. I think that we can break compatibility here to make the code more consistent